astropy/coordinates/baseframe.py

```diff
@@ -424,39 +424,121 @@ def __init__(self, *args, copy=True, representation_type=None,
         representation_type = kwargs.pop('representation_type', representation_type)
 
         if representation_type is not None or differential_type is not None:
+            representation_type, differential_type = self._infer_representation_differential_type(
+                representation_type, differential_type)
+            self.set_representation_cls(representation_type, **differential_type)
+        else:
+            self._representation = {'base': self.default_representation,
+                                    's': self.default_differential}
 
-            if representation_type is None:
```

Comment:
The diff is a bit messed up, but this line up to the next comment was moved to `_infer_representation_differential_data`

Reply:
(Edit: typo)